packages/core/test/renderToCanvas_gpu_test.js

```diff
@@ -0,0 +1,162 @@
+import * as cornerstone3D from '../src/index';
+import * as testUtils from '../../../utils/test/testUtils';
+import * as renderToCanvas_gpu_setStack from './groundTruth/renderToCanvas_gpu_setStack.png';
+import * as renderToCanvas_gpu_canvas from './groundTruth/renderToCanvas_gpu_canvas.png';
+import * as renderToCanvas_gpu_setStack_color from './groundTruth/renderToCanvas_gpu_setStack_color.png';
+import * as renderToCanvas_gpu_canvas_color from './groundTruth/renderToCanvas_gpu_canvas_color.png';
+
+const { cache, RenderingEngine, utilities, imageLoader, metaData, Enums } =
+  cornerstone3D;
+
+const { Events, ViewportType, InterpolationType } = Enums;
+
+const { fakeImageLoader, fakeMetaDataProvider, compareImages } = testUtils;
+
+const renderingEngineId = utilities.uuidv4();
+
+const viewportId = 'VIEWPORT';
+
+const AXIAL = 'AXIAL';
+
+function createViewport(renderingEngine, orientation, width, height) {
+  const element = document.createElement('div');
+
+  element.style.width = `${width}px`;
+  element.style.height = `${height}px`;
+  document.body.appendChild(element);
+
+  renderingEngine.setViewports([
+    {
+      viewportId: viewportId,
+      type: ViewportType.STACK,
+      element,
+      defaultOptions: {
+        background: [1, 0, 1], // pinkish background
+      },
+    },
+  ]);
+  return element;
+}
+
+describe('renderToCanvas -- GPU', () => {
+  beforeAll(() => {
+    window.devicePixelRatio = 1;
+    cornerstone3D.setUseCPURendering(false);
+  });
+  beforeEach(function () {
+    cache.purgeCache();
+    this.DOMElements = [];
+
+    this.renderingEngine = new RenderingEngine(renderingEngineId);
+    imageLoader.registerImageLoader('fakeImageLoader', fakeImageLoader);
+    metaData.addProvider(fakeMetaDataProvider, 10000);
+  });
+
+  afterEach(function () {
+    cache.purgeCache();
+    this.renderingEngine.destroy();
+    metaData.removeProvider(fakeMetaDataProvider);
+    imageLoader.unregisterAllImageLoaders();
+    this.DOMElements.forEach((el) => {
+      if (el.parentNode) {
+        el.parentNode.removeChild(el);
+      }
+    });
+  });
+
+  it('Should render one stack viewport of square size properly: nearest', function (done) {
```

Comment:
Ok stupid question time, but shouldn't the tests be marked async or something considering that there is a promise to `loadImageToCanvas` or maybe there is a magical `await` somewhere and I am simply none the wiser. 🤣

Reply:
No, the done does the same thing